readme.md

```diff
@@ -3,37 +3,58 @@
 [![Latest Stable Version](https://img.shields.io/badge/Latest%20Stable%20Version-1.0.0.Final-blue.svg)](http://search.maven.org/#search%7Cga%7C1%7Cg%3Aorg.mapstruct%20AND%20v%3A1.*.Final)
 [![Latest Version](https://img.shields.io/maven-central/v/org.mapstruct/mapstruct-processor.svg?maxAge=2592000&label=Latest%20Release)](http://search.maven.org/#search%7Cga%7C1%7Cg%3Aorg.mapstruct)
 
+* [What is MapStruct?](#what-is-mapstruct)
+* [Requirements](#requirements)
+* [Using MapStruct](#using-mapstruct)
+ * [Maven](#maven)
+ * [Gradle](#gradle)
+* [Documentation and getting help](#documentation-and-getting-help)
+* [Licensing](#licensing)
+* [Building from Source](#building-from-source)
+* [Links](#links)
+
 ## What is MapStruct?
 
-MapStruct is a Java [annotation processor](http://docs.oracle.com/javase/6/docs/technotes/guides/apt/index.html) for the generation of type-safe bean mapping classes.
+MapStruct is a Java [annotation processor](http://docs.oracle.com/javase/6/docs/technotes/guides/apt/index.html) for the generation of type-safe and performant mappers for Java bean classes.
+
+To create a mapping between two types, declare a mapper class like this:
+
+```java
+@Mapper
+public interface CarMapper {
 
-All you have to do is to define a mapper interface which declares any required mapping methods. During compilation, MapStruct will generate an implementation of this interface. This implementation uses plain Java method invocations for mapping between source and target objects, i.e. no reflection or similar.
+    CarMapper INSTANCE = Mappers.getMapper( CarMapper.class );
 
-Compared to writing mapping code from hand, MapStruct saves time by generating code which is tedious and error-prone to write. Following a convention over configuration approach, MapStruct uses sensible defaults but steps out of your way when it comes to configuring or implementing special behavior.
+    @Mapping(source = "numberOfSeats", target = "seatCount")
+    CarDto carToCarDto(Car car);
+}
+```
 
-Compared to dynamic mapping frameworks, MapStruct offers the following advantages:
+At compile time MapStruct will generate an implementation of this interface. The generated implementation uses plain Java method invocations for mapping between source and target objects, i.e. there is no reflection involved. By default, properties are mapped if they have the same name in source and target, but this and many other aspects can be controlled using `@Mapping` and a handful of other annotations.
```

Comment:
Why should we differentiate between "Java 6 compatible method invocations" and "Java method invocations"? 🤔